### PR TITLE
feat(webapp): map

### LINKED ISSUE #5807: 🚀 [Feature] Geolocation – Map To Show Locations Of Users And Groups
## 🚀 Feature

Geolocation – Map To Show Locations Of Users And Groups.

```gherkin
Feature: Map for users and groups
  As a user
  Do I want to find other users and groups in my area or in other places I travel to or the like.
```

### Todo

- [x] we like to have an additional page with a map
  - [x] map shall be reached by globus in header menu
    - [x] the one of the language menu
      - [ ] in the size of the avatar image
      - [x] tooltip:
        - de: `Landkarte anzeigen`
        - en: `Show map`
    - [ ] do we use a different icon in the language menu then?
- [x] chose map service:
  - [x] services:
    - [x] MapBox
      - @ulfgebhardt @ogerly @Tirokk decided for the package `v-mapbox`: https://github.com/geospoc/v-mapbox
    - [ ] Open Street Map
  - [x] comparison:
    - [x] https://www.rst.software/blog/google-maps-vs-mapbox-vs-openstreetmap-which-map-api-should-you-choose-and-why
- [x] have a search field above the map to enter the location to be shown
  - [x] search field is now in the map
    - [ ] label:
      - [ ] de: `Suchort oder -region eingeben`
      - [ ] en: `Enter search city or region`
  - [x] the location shall start at my own location
    - [x] if I have not entered my own location in the settings what happens then?
      - we start in Germany now and change that in the future
- [x] the map shall show flags or pins in different colors:
  - [x] users
  - [x] groups
- [x] map functionallities:
  - [x] zoom in and out
  - [x] move map with mouse
  - [x] hover over flag shows popup with name of user or group
    - [x] click routes to user or group profile
    - [x] similar to https://www.kartevonmorgen.org/
- [x] (see follow up issue #5901 ) map visibility (for later):
  - [x] even if a user is not logged in somehow defined users and groups can be seen on the map, followup see: #5928
- [x] if I don't want to be visible on a map as user
  - [x] I should not enter my location in the settings?
    - [x] may have a hint in the settings that by entering my location I a gree to show my location on the map
    - [x] and maybe hint that the position is not very specific
  - [x] (nope) or do I have to have a checkbox in the settings?
- [x] (see follow up issue #5901 ) to make the position a little more precise, especially in large cities, create the possibility of entering the postal code or a district of the city

### Issues

- relates #4546

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Basic implementation of `v-mapbox`- got thinks working

### Issues

- fixes #5807
- relates #4546

### Todo

- [x] have map icon on header
  - [x] map shall be reached by globus in header menu
    - [x] the one of the language menu
      - [ ] in the size of the avatar image
      - [x] tooltip:
        - de: `Landkarte anzeigen`
        - en: `Show map`
    - [x] (no) do we use a different icon in the language menu then?
- [x] may have a hint in the settings that by entering my location I a gree to show my location on the map
  - [x] and maybe hint that the position is not very specific
- [x] do we need the package `@mapbox/mapbox-gl-language`?
  - [x] tried, but does not work yet
- [x] implement `MAPBOX_TOKEN` in webapp config
  - [x] switch map off if there is no `MAPBOX_TOKEN`?
    - [x] header icon
    - [x] page
  - [x] implement it in docker compose and deployment all over the place
- [x] translate several texts
- [x] map optimization: see #5901
  - [ ] have a legend?
- [x] we need Node v16.5.0 or higher in webapp for the package `v-mapbox@legacy`
  - [x] we use Node v16.19.0, see #5842 
    - Node v19.3.0 was not working
- [x] mention the Node version in webapp README?
  - [x] have the latest version? See issue #5902
- [x] rebranding
  - changed map style buttons from `<ds-button>` to `<base-button>`
    - [x] adjust look of `inactive` button
    - [ ] try to implement `inactive` state into `BaseButton.vue`
      - !!! the map style inactive buttons don't have the right color
- [x] tests, see followup issue #5927
  - [x] decrease coverage to `61%` in webapp for now
- [x] cleanup: search for `Wolle`

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "@babel/plugin-syntax-dynamic-import": "^7.8.3",
     "@babel/preset-env": "~7.9.0",
     "@faker-js/faker": "5.1.0",
-    "@nuxtjs/composition-api": "0.33.1",
+    "@nuxtjs/composition-api": "0.32.0",
```
Comment (Tirokk) [inline webapp/package.json:72]: Thanks for your work @ulfgebhardt
I have tried that as well and got a different error messages.

I try again.

File: webapp/graphql/groups.js [hunk 2]
```diff
@@ -171,8 +174,11 @@ export const groupQuery = (i18n) => {
           url
         }
         locationName
+        # ...location
         location {
```
Comment (Tirokk) [inline webapp/graphql/groups.js:178]: Fragment `location` should not have the same name as propety `location` I guess.
Sometime we have a GQL error seems related to this, see issue.

File: webapp/utils/utils.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const objectValuesToArray = (obj) => {
```
Comment (Mogge) [inline webapp/utils/utils.js:1]: This is https://docs-lodash.com/v4/to-array/
Please import it from there

**Review by Mogge — CHANGES_REQUESTED** [review]
Some smaller things to start with

**Review by Mogge — APPROVED** [review]

**Review by Mogge — APPROVED** [review]

**Pull request merged**